beetsplug/alternatives.py

```diff
@@ -359,6 +411,25 @@ def parse_config(self, config):
 
         super(SymlinkView, self).parse_config(config)
 
+    def is_in_external(self, path):
+        return os.path.islink(syspath(path))
+
+    def item_change_actions(self, item, path, dest):
+        """ Returns the necessary actions for items that were previously in the
+        external collection, but might require metadata updates.
+        """
+        actions = []
+
+        if not _samelink(path, dest):
+            # The path of the link itself changed
+            actions.append(self.MOVE)
+        elif not util.samefile(path, item.path):
+            # link target changed
+            actions.append(self.MOVE)
+
+        # FIXME: test album art link
```

Comment:
We don’t need to fix this, it is is already working as expected :tada:. Since we’re only linking the album art is always properly embedded in the target file.

Reply:
What I meant to say is that embedding art in SymlinkViews probably should not be supported in the first place. There's already the embedart plugin for your beets library, and no need for (potentially many) alternative collections to attempt to embed art in these files _again_. Rather, it might be interesting to symlink the artwork itself in the alt collection, in particular if no embedding is intended. That'd be a new feature, however.

Reply:
I'm going to remove this and open a new issue to keep track of this feature (I'm not really interested myself, since I don't use SymlinkViews).

Reply:
See #53